src/Core/Module/ModuleRepository.php

```diff
@@ -175,19 +175,49 @@ public function setActionUrls(ModuleCollection $collection): ModuleCollection
         return $this->adminModuleDataProvider->setActionUrls($collection);
     }
 
-    public function clearCache(?string $moduleName = null): bool
+    /**
+     * @param string|null $moduleName The module to clear the cache for. If the name is null, the cache will be cleared for all modules.
+     * @param bool $allShops Default to false. If the value is true, the cache will be cleared for all the active shops. If not it will be cleared only for the shop in the context.
+     *
+     * @return bool
+     */
+    public function clearCache(?string $moduleName = null, bool $allShops = false): bool
     {
         $this->installedModules = null;
-        if ($moduleName !== null && $this->cacheProvider->contains($this->getCacheKey($moduleName))) {
-            return $this->cacheProvider->delete($this->getCacheKey($moduleName));
+        if ($moduleName !== null) {
+            if ($allShops) {
+                foreach (Shop::getShops(true, null, true) as $shopId) {
+                    $cacheKey = $this->getCacheKey($moduleName, $shopId);
+                    if ($this->cacheProvider->contains($cacheKey)) {
+                        if (!$this->cacheProvider->delete($cacheKey)) {
+                            return false;
+                        }
+                    }
+                }
+
+                return true;
+            } else {
+                $cacheKey = $this->getCacheKey($moduleName);
+                if ($this->cacheProvider->contains($cacheKey)) {
+                    return $this->cacheProvider->delete($cacheKey);
+                }
+            }
         }
 
         return $this->cacheProvider->deleteAll();
     }
 
-    private function getCacheKey(string $moduleName): string
+    /**
+     * @param string $moduleName
+     * @param int|null $shopId If this parameter is given, the key returned will be the one for the shop. Otherwise, it will be the cache key for the shop in the context.
+     *
+     * @return string
+     */
+    private function getCacheKey(string $moduleName, ?int $shopId = null): string
```

Comment:
```suggestion
    protected function getCacheKey(string $moduleName, ?int $shopId = null): string
```

Reply:
What is the reason to add `protected` ?

Reply:
That allows people to override `clearCache` and use this method in this one. Else devs can't correctly override `clearCache` method.